server/models/chan.ts

```diff
@@ -63,20 +62,22 @@ class Chan {
 	static optionalProperties = ["userAway", "special", "data", "closed", "num_users"];
 
 	constructor(attr?: Partial<Chan>) {
-		_.defaults(this, attr, {
-			id: 0,
-			messages: [],
-			name: "",
-			key: "",
-			topic: "",
-			type: ChanType.CHANNEL,
-			state: ChanState.PARTED,
-			firstUnread: 0,
-			unread: 0,
-			highlight: 0,
-			users: new Map(),
-			muted: false,
-		});
+		this.id = 0;
+		this.messages = [];
+		this.name = "";
+		this.key = "";
+		this.topic = "";
+		this.type = ChanType.CHANNEL;
+		this.state = ChanState.PARTED;
+		this.firstUnread = 0;
+		this.unread = 0;
+		this.highlight = 0;
+		this.users = new Map();
+		this.muted = false;
```

Comment:
Since this is a class now, why don't we assign all of these as defaults on the class properties instead of setting them in the constructors?